pkg/serverstate/statetest/test_runner_ondemand.go

```diff
@@ -34,6 +34,28 @@ func TestOnDemandRunnerConfig(t *testing.T, factory Factory, restartF RestartFac
 		require.Equal(codes.NotFound, status.Code(err))
 	})
 
+	t.Run("Set with no name", func(t *testing.T) {
+		require := require.New(t)
+
+		s := factory(t)
+		defer s.Close()
+
+		// Set
+		rec := serverptypes.TestOnDemandRunnerConfig(t, &pb.OnDemandRunnerConfig{
+			OciUrl:               "h/w:s",
+			EnvironmentVariables: map[string]string{"CONTAINER": "DOCKER", "FOO": "BAR"},
+			TargetRunner:         &pb.Ref_Runner{Target: &pb.Ref_Runner_Any{Any: &pb.Ref_RunnerAny{}}},
+			PluginConfig:         []byte(`{"foo":"bar"}`),
+			ConfigFormat:         pb.Hcl_JSON,
+			Default:              true,
+		})
+
+		result, err := s.OnDemandRunnerConfigPut(rec)
+		require.NoError(err)
+		require.NotEmpty(result.Id)
+		require.NotEmpty(result.Name)
```

Comment:
Maybe one more for extra assurance:

```golang
require.Equal(result.Name, result.Id)
```

Reply:
I thought about this, but it feels like testing an implementation detail. We don't really care what the name is - just that we gave it _some_ name.

I honestly don't love using the ulid as the name. It's very long, and it's hard for human eyes to distinguish from other ulids. I feel like i'd prefer us to use pet names, or 6 random letters k8s-pod-hash-style, or something.